x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -658,10 +678,26 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
             }
         } catch (Exception e) {
             if (ignoreMalformed) {
-                if (subParser != null) {
+                if (malformedContentForSyntheticSource != null) {
+                    if (subParser != null) {
+                        XContentHelper.drainParser(subParser, malformedContentForSyntheticSource);
+                    } else {
+                        // We don't use DrainingXContentParser since we don't want to go beyond current field
+                        malformedContentForSyntheticSource.copyCurrentStructure(context.parser());
+                    }
+
+                    int parentOffset = context.parent() instanceof RootObjectMapper ? 0 : context.parent().fullPath().length() + 1;
+                    var nameValue = new IgnoredSourceFieldMapper.NameValue(
+                        name(),
+                        parentOffset,
+                        XContentDataHelper.encodeXContentBuilder(malformedContentForSyntheticSource)
+                    );
+                    context.addIgnoredField(nameValue);
```

Comment:
I'd think the section above (and most of the other changes) will be reused in other cases too, like geopoint. It may be worth encapsulating in a class to test it in isolation too:
 - MalformedContentHelper.init(XContentBuilder builder)
 - MalformedContentHelper.addField(XContentBuilder builder, String fieldName)
 - MalformedContentHelper.addValue(XContentBuilder builder, Number value)
 - MalformedContentHelper.finalizeAndStore(XContentBuilder builder, DocumentParserContext context, XContentParser subparser)

Reply:
I like that suggestion but i am not sure how API should look like. I looked at geo point briefly and parsing logic there is pretty complex, i am not sure it will map exactly to current usage pattern. I'll introduce a generic helper together with support for next complex feild.